source/particle/property/grain_size.cc

```diff
@@ -0,0 +1,165 @@
+/*
+  Copyright (C) 2022 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file LICENSE.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/property/grain_size.h>
+#include <aspect/material_model/grain_size.h>
+#include <aspect/initial_composition/interface.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      template <int dim>
+      GrainSize<dim>::GrainSize ()
+        :
+        material_inputs(1,0),
+        material_outputs(1,0)
+      {}
+
+
+
+      template <int dim>
+      void
+      GrainSize<dim>::initialize ()
+      {
+        material_inputs  = MaterialModel::MaterialModelInputs<dim>(1, this->n_compositional_fields());
+        material_outputs = MaterialModel::MaterialModelOutputs<dim>(1, this->n_compositional_fields());
+
+        AssertThrow(Plugins::plugin_type_matches<const MaterialModel::GrainSize<dim>>(this->get_material_model()),
```

Comment:
Why this limitation? I don't think you access anything directly from this material model, or do you? What if I want to implement a slight variation of that class by copying it?